File: backend/models/Damnificado.js
const mongoose = require('mongoose');

const damnificadoSchema = new mongoose.Schema({
  inmueble: { type: mongoose.Schema.Types.ObjectId, ref: 'Inmueble', required: true },
  nombre: { type: String, default: '' },
  edad: { type: Number, default: 0 },
  sexo: { type: String, enum: ['M', 'F', ''] },
  tipo_identificacion: { type: String, default: '' },
  numero_identificacion: { type: String, default: '' },
  estado: {
    type: String,
    enum: ['fallecido', 'lesionado_grave', 'lesionado_leve', 'ileso'],
    default: 'ileso',
  },
  requiere_traslado: { type: Boolean, default: false },
  observaciones: { type: String, default: '' },
  sincronizado: { type: Boolean, default: true },
  creado_en: { type: Date, default: Date.now },
  actualizado_en: { type: Date, default: Date.now },
});

damnificadoSchema.pre('save', function (next) {
  this.actualizado_en = Date.now();
  next();
});

module.exports = mongoose.model('Damnificado', damnificadoSchema);

File: mobile/lib/models/damnificado.dart
class Damnificado {
  final String id;
  final String inmuebleId;
  final String nombre;
  final int edad;
  final String sexo;
  final String tipoIdentificacion;
  final String numeroIdentificacion;
  final String estado;
  final bool requiereTraslado;
  final String observaciones;
  final bool sincronizado;

  Damnificado({
    required this.id,
    required this.inmuebleId,
    this.nombre = '',
    this.edad = 0,
    this.sexo = '',
    this.tipoIdentificacion = '',
    this.numeroIdentificacion = '',
    this.estado = 'ileso',
    this.requiereTraslado = false,
    this.observaciones = '',
    this.sincronizado = false,
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'inmuebleId': inmuebleId,
        'nombre': nombre,
        'edad': edad,
        'sexo': sexo,
        'tipoIdentificacion': tipoIdentificacion,
        'numeroIdentificacion': numeroIdentificacion,
        'estado': estado,
        'requiereTraslado': requiereTraslado ? 1 : 0,
        'observaciones': observaciones,
        'sincronizado': sincronizado ? 1 : 0,
      };

  factory Damnificado.fromMap(Map<String, dynamic> map) => Damnificado(
        id: map['id'] as String,
        inmuebleId: map['inmuebleId'] as String,
        nombre: map['nombre'] as String? ?? '',
        edad: map['edad'] as int? ?? 0,
        sexo: map['sexo'] as String? ?? '',
        tipoIdentificacion: map['tipoIdentificacion'] as String? ?? '',
        numeroIdentificacion: map['numeroIdentificacion'] as String? ?? '',
        estado: map['estado'] as String? ?? 'ileso',
        requiereTraslado: (map['requiereTraslado'] as int? ?? 0) == 1,
        observaciones: map['observaciones'] as String? ?? '',
        sincronizado: (map['sincronizado'] as int? ?? 0) == 1,
      );

  Map<String, dynamic> toJson() => {
        'nombre': nombre,
        'edad': edad,
        'sexo': sexo,
        'tipo_identificacion': tipoIdentificacion,
        'numero_identificacion': numeroIdentificacion,
        'estado': estado,
        'requiere_traslado': requiereTraslado,
        'observaciones': observaciones,
      };
}
